pkg/envoy/proxy.go

```diff
@@ -54,6 +55,46 @@ func (p *Proxy) HasPodMetadata() bool {
 	return p.PodMetadata != nil
 }
 
+// StatsHeaders returns the headers required for SMI metrics
+func (p *Proxy) StatsHeaders() map[string]string {
+	unknown := "unknown"
+	podName := unknown
+	podNamespace := unknown
+	podControllerKind := unknown
+	podControllerName := unknown
+
+	if p.PodMetadata != nil {
+		if len(p.PodMetadata.Name) > 0 {
+			podName = p.PodMetadata.Name
+		}
+		if len(p.PodMetadata.Namespace) > 0 {
+			podNamespace = p.PodMetadata.Namespace
+		}
+		if len(p.PodMetadata.WorkloadKind) > 0 {
+			podControllerKind = p.PodMetadata.WorkloadKind
+		}
+		if len(p.PodMetadata.WorkloadName) > 0 {
+			podControllerName = p.PodMetadata.WorkloadName
+		}
+	}
+
+	if podControllerKind == "ReplicaSet" {
+		// Assume ReplicaSets are controlled by a Deployment unless their names
+		// do not contain a hyphen.
```

Comment:
What's the need for this assumption?

Reply:
This assumption is already being made in the current prometheus config:
https://github.com/openservicemesh/osm/blob/ad48e021e597484193ed38e3ae95c91f93fa87c3/charts/osm/templates/prometheus-configmap.yaml#L102-L112

I suppose we could add some sort of recursive lookup to find the ultimate controller of each pod in the injector, but not sure if that's worth it.

Reply:
Let's drop a comment referencing the prometheus config assumption. It will be clearer as to why this assumption is being made in `proxy.go`.

Reply:
Sounds good, fixed.